Library/Homebrew/cask/cask.rb

```diff
@@ -81,6 +81,21 @@ def versions
                           .reverse
     end
 
+    def os_versions
+      @os_versions ||= begin
+        version_os_hash = {}
+        actual_version = MacOS.full_version.to_s
+
+        MacOS::Version::SYMBOLS.each do |os_name, os_version|
+          MacOS.full_version = os_version
+          version_os_hash[os_name] = CaskLoader.load(token).version
+        end
+
+        MacOS.full_version = actual_version
```

Comment:
Maybe want this in an `ensure` so that it doesn't mess up e.g. debugging information on an exception.